packages/contracts-bedrock/contracts/test/Hashing.t.sol

```diff
@@ -21,19 +21,20 @@ contract Hashing_Test is CommonTest {
     }
 
     function test_hashCrossDomainMessage_differential(
-        uint256 _nonce,
+        uint240 _nonce,
+        uint16 _version,
         address _sender,
         address _target,
         uint256 _value,
         uint256 _gasLimit,
         bytes memory _data
     ) external {
-        // Discard any fuzz tests with an invalid version
-        (, uint16 version) = Encoding.decodeVersionedNonce(_nonce);
-        vm.assume(version < 2);
+        // Ensure the version is valid
```

Comment:
@tynes can you check if this looks good?

Reply:
Yeah, it's worth expanding on what it does because it's cool. 

The code on the left generates random nonces, and then throws out anything where the version number is not either 0 or 1. So if the inputs are naively random, then theoretically we'd be throwing out all but 2 out of every 65536 inputs. 

The code on the right will take the randomly generated `uint16 version`, and modulo it into the range `[0,1]`.